### PR TITLE
Anagram String Code

### LINKED ISSUE #1969: Anagram String
## 🚀 Feature

(A clear and concise description of what the feature is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes

## Motivation

Anagram is frequently aksed and used cp problem.

## Pitch
An anagram is a word or phrase formed by rearranging the letters of a different word or phrase, typically using all the original letters exactly once

### PR DESCRIPTION
**Pull request by RuchiGitHu**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

An anagram of a string is another string that contains the same characters, only the order of characters can be different.
For example, “abcd” and “dabc” are an anagram of each other.

### Checklist

- [X] I've read the contribution guidelines.
- [X] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixed #1969. I have linked my code in readme file and also made the changes required for pull request.

### CODE REVIEW

**Review by atarax665 — COMMENTED** [review]
Please rename the file from ```Anagram_Java.java``` to ```anagram.java```. I will review the code once the coding period starts until then you can look into some other approaches to check if two strings are anagram. 
Happy Coding 👍

File: Java/cp/Anagram.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/* This is a java code of checking whether the Strings are anagram to each outher and not...
+An anagram of a string is another string that contains the same characters, only the order of characters can be different.
+For example, “abcd” and “dabc” are an anagram of each other.
+*/
```
Comment (HarshCasper) [inline Java/cp/Anagram.java]: Make the beginner pitch more descriptive and justify the content. You can also discuss your approach herein

File: Java/cp/Anagram.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	public static void main(String[] args) {
+
+		Scanner scan = new Scanner(System.in);
+		System.out.println("Please enter the first String : ");
+		String a = scan.next();
+		System.out.println("Please enter the Second String : ");
+		String b = scan.next();
+		scan.close();
+		boolean ret = isAnagram(a, b);
+		System.out.println((ret) ? "String are Anagrams..." : "Strings are Not Anagrams...");
```
Comment (HarshCasper) [inline Java/cp/Anagram.java:50]: Use proper console statements.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny stylistic changes requested. Code looks good to merge :+1:

File: Java/cp/Anagram.java (hunk 2 above)
Comment (RuchiGitHu) [reply Java/cp/Anagram.java:41]: Hi @HarshCasper , Please let me know, what you mean by "Proper Console Statement". I didn't understand that. 
Thanks for your review...

Comment (RuchiGitHu) [reply Java/cp/Anagram.java:50]: Thanks, Just now tried to made it as required..

**Review by atarax665 — APPROVED** [review]

File: Java/cp/Anagram.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		return c;   //returning boolean value from function.
+	}
+	public static void main(String[] args) {   // The main function
+		Scanner scan = new Scanner(System.in); // Assigning variable 'Scan' for taking input from user.
+		System.out.println("Please enter the first String : ");
+		String a = scan.next();  // Storing value of first String taken from 'scan' to String a.
+		System.out.println("Please enter the Second String : ");
+		String b = scan.next();
+		scan.close(); // Close to receive input from user.
+		boolean ret = isAnagram(a, b); // Calling function in Annagram class.
```
Comment (HarshCasper) [inline Java/cp/Anagram.java]: Use Comments above the line of code, not beside it. 

And please skip out redundant comments. They do no good and make the code look un-clean and at some places forced as well

File: Java/cp/Anagram.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+* For example, “abcd” and “dabc” are an anagram of each other.
+*/
+import java.util.Scanner;
+import java.lang.String;
+public class Annagram {
+/* Called function(isAnagram) from main function
+* to check whether the Strings are Anagrams.
+*/
+	static boolean isAnagram(String a, String b) {
+ // Changing all the charactor of a into Lower case.
```
Comment (HarshCasper) [inline Java/cp/Anagram.java]: Indent the Comments as well

File: Java/cp/Anagram.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		a = a.toLowerCase();
+		b = b.toLowerCase();
+// the function has boolean type so we have to return the boolean value.
```
Comment (HarshCasper) [inline Java/cp/Anagram.java]: Indent the Comments as well

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
Remove the NeoAlgo Submodule

**Comment by RuchiGitHu** [comment]
> Remove the NeoAlgo Submodule

Please help me, to remove this submodule as it pushed mistakenly. Please guide me, how can I remove NeoAlgo submodule....
Thanks for your guidance...

**Comment by HarshCasper** [comment]
Refer [this](https://gist.github.com/myusuf3/7f645819ded92bda6677) and [this](https://stackoverflow.com/questions/1260748/how-do-i-remove-a-submodule)

**Comment by RuchiGitHu** [comment]
Please review #2843, my new pull request for same issue.
Thanks @HarshCasper @atarax665 for your support.

**Pull request closed without merging**